icu4c/source/common/uloc_tag.cpp

```diff
@@ -2092,12 +2092,13 @@ ultag_parse(const char* tag, int32_t tagLen, int32_t* parsedLen, UErrorCode* sta
             int32_t oldTagLength = tagLen;
             if (tagLen < newTagLength) {
                 uprv_free(tagBuf);
-                tagBuf = (char*)uprv_malloc(newTagLength + 1);
+                // Change t->buf after the free and before return to avoid the second double free in
+                // the destructor of t when t is out of scope.
```

Comment:
I really think it'd be clearer to write `ultag_close()` instead of _the destructor_, if that's what you mean. It took me unnecessarily long time to figure out that this probably was what you meant.

Reply:
the function signature of ultag_close() is
```
void ultag_close(ULanguageTag* langtag);
```
and the only code which ever call ultag_close is by the destructor of LocalULanguageTagPointer
There are no code in ultag_parse call ultag_close directly so if you said it is free by ultag_close it is very difficult for the reader to understand WHERE is such call connect to THIS (ultag_parse) function. My updated comments said "second double free in the destructor of t " will bring the reader to look into the type of t and therefore look into
the desctructor of LocalULanguageTagPointer and then see it call the close function which is ultag_close. If my comments only mention ultag_close it is harder for the reader to understand why that ultag_close will be called.